### DISCUSSION TITLE
Ability to pin inspector view

### DISCUSSION THREAD

**Discussion by lentsius-bark** [godotengine/godot-proposals#4226] [Editor]
The idea for this is:
- there's a pin button in the inspector which locks the current inspector panel (top right)
- selecting other nodes does not update the inspector
- changing the values in the pinned inspector view applies them to the node that was selected when the pin was triggered
- unpinning the inspector makes it go back to updating according to the selected node

*this has the potential of solving quite the number of UX issues where changing values requires going back and forth between nodes and/or resources*

Thoughts?

**Comment by lentsius-bark** [comment]
Had a talk about this with @fire and it seems like a plausible improvement which does not change the current design of godot nor does it introduce a new element, 

here's a screenshot from blender:
![image](https://user-images.githubusercontent.com/38346332/158582000-63754dbf-63bd-4b73-a2f5-550fdf89f588.png)

**Reply by fire** [reply]
The best part about this proposal is it doesn't require a per property inspector change ui/coding wise. There is no duplication of the inspector panel code.

The drawer design requires a per inspector panel code duplication which can cause maintenance issues. The pin only requires the pinning toggle states and a generic system for any inspector panel.

![Drawers](
https://cdn.discordapp.com/attachments/443214210701197312/953610088277508096/FFzmkH1WQAQahRR.png)

**Reply by lentsius-bark** [reply]
agreed, the fact that this doesn't require any additional UI and makes use of the already existing elements to provide extra functionality allowing for new UX sounds good and makes sense!

**Comment by lentsius-bark** [comment]
having talked about this further with other godot users, here's additional use cases:
- you might want to keep the inspector of a blur shader pinned, while moving things underneath it, having a quick way to come back to it and tweak its values according to what the things below it look like
- drag and dropping nodes, i've had my team go crazy over the fact that trying to drag and drop nodes, changes the inspector where they were trying to drop them
- tweaking particle properties as you play around with the objects' positions in the scene

**Reply by Calinou** [reply]
> drag and dropping nodes, i've had my team go crazy over the fact that trying to drag and drop nodes, changes the inspector where they were trying to drop them

This can be fixed by making the inspector switch nodes when *releasing* the mouse button in the scene tree dock instead of when *pressing* the mouse button. For reference, this is what Unity does already. It would also improve perceived performance when clicking on nodes in the Scene tree dock, as the visual feedback for selecting a node would always display immediately (even if the inspector takes a while to update).

**Reply by KoBeWi** [reply]
https://github.com/godotengine/godot/pull/61162

**Comment by eon-s** [comment]
The Animation Player bottom panel already have a pinning option but is limited to one player.

For this to be more useful, the editor will need to be able to open more than one panel of the same type (multiple inspectors, multiple filesystem, et cetera). otherwise this cannot be used to watch, compare and other uses that pinning has on similar tools.

**Comment by ScrewTSW** [comment]
One possible implementation suggestion for this feature.
My code is incomplete and doesn't handle some edge cases like Sub-Resource selector and the documentation button, the search input, but takes care of the main issue.
https://github.com/godotengine/godot/compare/master...ScrewTSW:pin-inspector-node-poc?expand=1
![inspector_pin_node_poc-2022-05-18_18 10 20](https://user-images.githubusercontent.com/16451875/169102066-113c3997-d4d3-474d-98d3-e9d3d0aca6ce.gif)

**Reply by fire** [reply]
Is this submitted as pull request with an associated proposal so it can be reviewed?

**Reply by lentsius-bark** [reply]
This is everything imagined it to be!
